### DISCUSSION TITLE
Docker UP

### DISCUSSION THREAD

**Discussion by jcgava** [savannabits/filament-tenancy-starter#3] [Q&A]
Hello friend.
Information on how to climb docker.
Thanks for sharing.

**Comment by coolsam726** [comment]
Hi @jcgava, 
Docker is set out of the box using docker compose.
1. Adjust the .env variables, paying attention to the following:
* **DOCKER_UID=1000** Set this to your current user's uid, which you can find by simply typing id on your terminal. This used when you want to bash into the app container and do stuff like `php artisan ...` and `composer install`
* `DOCKER_HOST_USER=dev` - The alias name under which the user with uid above will be created. You will use this when logging into the app container as follows: `docker compose exec -u dev app bash`
* `DOCKER_NETWORK_NAME=savannabits` - The containers use an external docker network to communicate. You will have to create this network using e.g `docker network create savannabits`. Give your network whichever name you want.
* `DOCKER_SERVER_EXPOSED_PORT`. This is the port that the your app will be running under when docker is up and running.
* `DOCKER_REDIS_EXPOSED_PORT` For the redis container
* `DB_HOST=thishost` - In this setup, the containers have been set to access the host OS using the hostname `thishost`. This is set in docker_compose.yml. If you would like to use a database hosted on your host, you will have to use `thishost` instead of `localhost`.
* The other variables are pretty straight forward.
2. After setting the variables, `cd your-project-directory/` and run `docker compose build app`
3. Once app is built, spin up the containers with `docker compose up -d` and you are up and running.

Out of the box you will get the following:
* app container which hosts the code. Whenever you want to run any artisan or composer command, you will have to run it inside the app container by bashing into the container using: `docker compose exec -u dev app bash`. Replace `dev` with the name of your DOCKER_HOST_USER variable.
*  A queue container to automatically run the queue worker.
* A cron container to automatically run the Laravel Scheduler CRON Job. Scheduled jobs will work out of the box.
* A server container to serve you app using nginx. If you would like to fine-tune your nginx config, adjust it under `.docker/config/appserver.conf.template`
* A websocket container configurable with laravel-websockets

**Comment by jcgava** [comment]
Thank you, I managed to run

**Comment by batuhanhazarguler** [comment]
I've followed the steps you provided, but I still can't get the application to run. Could you prepare a more detailed guide? Also, how can I run the project without Docker, and what is needed for that? Thank you for your assistance. The project doesn't load any CSS data when running without Docker; can you help with that? Thanks for everything; it looks like an amazing project.
